### ISSUE TITLE
Networking flags not added when running with sudo 

### ISSUE THREAD

**Issue by arthurfeeney** [singularityhub/singularity-compose#39]
Hello, I have realized that when I try to run singularity-compose, it does not use singularity's --net options. This happened with or without sudo. 
my setup works if I use a shell script to start each of the instances and manually include the ``--net`` flags.

For instance, if I run ``singularity-compose --log-level DEBUG up`` (I would usually have to put in my password), the logs show that it is not using ``--net`` or setting the IPs and ports. Each instance just looked like this:

~~~~
DEBUG singularity instance start --bind some_binding --bind etc.hosts --bind resolv.conf db.sif db
~~~~

As an example, the postgres instance I am using looks like

~~~~
version: "1.0"
instances:
  db:
     build:
       contenxt: ./
       reciple: /path/to/recipe
    ports:
       - "5432:5432"
    volumes:
       - ./yadayada:/blahblah
~~~~

If the containers try to communicate with each other, they won't find each other and it will fail.

I also tried explicitly running it with sudo like ``sudo /path/to/singularity-compose up`` and the result was the same.

Is there something obvious that I am missing with how I need to use sudo? It seems like the networking flags should be getting included automatically since I am using ports...? :thinking: 

#### Workaround:

One fix I found was when I removed the ``if sudo:`` wrapping the network options inside Instance.create (This seems to have been added in #27), it works fine and includes the needed --net commands. Everything works perfectly after that!
I suspect that this simple workaround introduces some other issues, but I just thought it may be helpful to point out where the issue seems to be occurring for me. I might also have some time to dig into this more over the weekend.

#### Version Info:
- singularity is 3.7
- singularity-compose was built from master fairly recently.

Thanks!

**Comment by vsoch** [comment]
This must be a bug, and thank you for pointing this out! Would you care to open a PR to fix it, as you've found?

**Comment by vsoch** [comment]
@pierlauro could you please take a look?

**Comment by pierlauro** [comment]
There should be no side effect with removing that sudo check. I recall it was not possible for non-sudoers to use bridged networking some time ago, so it was kind of a "sanity check": don't create hosts **since you can't use them**. As now that seems to be possible from @arthurfeeney's message, that check can go away :)

**Comment by vsoch** [comment]
Great! @arthurfeeney would you care to open a PR to remove the check?

**Comment by vsoch** [comment]
Closed by #40
